envision/client.py

```diff
@@ -73,9 +75,18 @@ def __init__(
         num_retries: int = 5,
         wait_between_retries: float = 0.5,
         output_dir: str = None,
+        client_id: str = None,
     ):
         self._log = logging.getLogger(self.__class__.__name__)
-        client_id = str(uuid.uuid4())[:8]
+
+        current_time = datetime.now().strftime("%Y%m%d%H%M%S%f")[:-4]
+        if client_id:
+            # String length limit for display is 20 characters
+            # Replace all special (non-alphanumeric) characters to "_" to avoid invalid key values
+            client_id = re.sub("\W+", "_", client_id[:20]) + "_"
+            client_id += current_time
```

Comment:
Can we use a string formating here? `client_id = f"{client_id}_{current_time}"`. Then you get rid of the `_` suffix and string concatenation. Need to replace the `client_id` if you rename the var.